src/storage/test/IndexTest.cpp

```diff
@@ -0,0 +1,670 @@
+        }
+    }
+    return indexes;
+}
+
+TEST(IndexTest, InsertVerticesTest) {
+    fs::TempDir rootPath("/tmp/InsertVerticesTest.XXXXXX");
+    std::unique_ptr<kvstore::KVStore> kv = TestUtils::initKV(rootPath.path());
+    LOG(INFO) << "Prepare meta...";
+    auto indexes = mockIndexes(false, 3, 3);
+    auto schemaMan = mockSchemaMan(0, indexes, std::vector<cpp2::IndexItem>());
+    cpp2::AddVerticesRequest req;
+    req.space_id = 0;
+    req.overwritable = true;
+    for (auto partId = 0; partId < 3; partId++) {
+        std::vector<cpp2::Vertex> vertices;
+        for (auto vertexId = partId * 10; vertexId < 10 * (partId + 1); vertexId++) {
+            std::vector<cpp2::Tag> tags;
+            for (auto tagId = 3001; tagId < 3010; tagId++) {
+                RowWriter writer;
+                for (int64_t numInt = 0; numInt < 3; numInt++) {
+                    writer << numInt;
+                }
+                for (auto numString = 3; numString < 6; numString++) {
+                    writer << folly::stringPrintf("tag_string_col_%d", numString);
+                }
+                auto val = writer.encode();
+
+                tags.emplace_back(apache::thrift::FragileConstructor::FRAGILE,
+                                  tagId,
+                                  std::move(val));
+            }
+            vertices.emplace_back(apache::thrift::FragileConstructor::FRAGILE,
+                                  vertexId,
+                                  std::move(tags));
+        }
+        req.parts.emplace(partId, std::move(vertices));
+    }
+    auto* processor = AddVerticesProcessor::instance(kv.get(), schemaMan.get(), nullptr);
+    auto fut = processor->getFuture();
+    processor->process(req);
+    auto resp = std::move(fut).get();
+    EXPECT_EQ(0, resp.result.failed_codes.size());
+    LOG(INFO) << "Check data...";
+    for (auto partId = 0; partId < 3; partId++) {
+        for (auto vertexId = 10 * partId; vertexId < 10 * (partId + 1); vertexId++) {
+            auto prefix = NebulaKeyUtils::vertexPrefix(partId, vertexId);
+            std::unique_ptr<kvstore::KVIterator> iter;
+            EXPECT_EQ(kvstore::ResultCode::SUCCEEDED, kv->prefix(0, partId, prefix, &iter));
+            int32_t rowCount = 0;
+            while (iter->valid()) {
+                rowCount++;
+                iter->next();
+            }
+            EXPECT_EQ(9, rowCount);
+        }
+    }
+    LOG(INFO) << "Check index...";
+    for (auto partId = 0; partId < 3; partId++) {
+        for (auto indexId = 3001; indexId < 3010; indexId++) {
+            auto prefix = NebulaKeyUtils::indexPrefix(partId, indexId);
+            std::unique_ptr<kvstore::KVIterator> iter;
+            EXPECT_EQ(kvstore::ResultCode::SUCCEEDED, kv->prefix(0, partId, prefix, &iter));
+            int32_t rowCount = 0;
+            while (iter->valid()) {
+                rowCount++;
+                iter->next();
+            }
+            EXPECT_EQ(10, rowCount);
```

Comment:
Why we will have one more index row?

Reply:
Aha, Let's do the math：
**3part * 10vertex * 9tag = 270(row)**
```
for (auto partId = 0; partId < 3; partId++) {
        for (auto vertexId = 10 * partId; vertexId < 10 * (partId + 1); vertexId++) {
            EXPECT_EQ(9, rowCount);
        }
    }
```

**3part * 9index(same with tag num) * 10 vertex = 270(row)**
```
    for (auto partId = 0; partId < 3; partId++) {
        for (auto indexId = 3001; indexId < 3010; indexId++) {
            EXPECT_EQ(10, rowCount); <<-- this is means 10 vertices.
        }
    }
```

Reply:
My bad, got it.